src/cursor_manager.ts

```diff
@@ -528,6 +534,58 @@ export class CursorManager implements Disposable {
         window.activeTextEditor.selections = selections;
     }
 
+    private async addSelectionToFindMatch(type: "prev" | "next") {
+        const editor = window.activeTextEditor;
+        if (!editor) return;
+        await this.main.cursorManager.waitForCursorUpdate(editor);
+        const run = () =>
+            commands.executeCommand(`editor.action.addSelectionTo${type == "prev" ? "Previous" : "Next"}FindMatch`);
+
+        if (this.main.modeManager.isInsertMode) {
+            await run();
+            return;
+        }
+
+        if (editor.selection.isEmpty) {
+            await run();
+            const selections = editor.selections;
+            this.main.client.input("<Esc>"); // Correcting the position
```

Comment:
I don't love the delays. Maybe you can await the input and then await a cursor update.

Reply:
> await the input

I just discovered that "input" can be awaited. :smile: